### PR TITLE
Fixed version parsing issue under certain circumstances

### LINKED ISSUE #107: `VERSION_RE` fails to match version info from `mysql` installed with `brew` on OSX High Sierra
### What action do you want to perform
Run tests with `mysql_proc` fixture.

OS: OS X High Sierra
Python Version: 3.6.1
MySQL Version: 5.7.23 (Homebrew)
pytest-mysql Version: 2.0.0

version output:
```
$mysqld --version
mysqld  Ver 5.7.23 for osx10.13 on x86_64 (Homebrew)
```

### What are the results
`VERSION_RE.match(version_output)` returns `None` where version number `str` is expected.
```
self = <pytest_mysql.executor.MySQLExecutor: "
        /..." 0x104790fd0>

    def version(self):
        """Read MySQL's version."""
        version_output = subprocess.check_output(
            [self.mysqld, '--version']
        ).decode('utf-8')
>       return self.VERSION_RE.match(version_output).groupdict()['version']
E       AttributeError: 'NoneType' object has no attribute 'groupdict'

../../../.pyenv/versions/mmt-server-promotion2/lib/python3.6/site-packages/pytest_mysql/executor.py:80: AttributeError
```

### What are the expected results
Version number is parsed and tests run without errors.

### PR DESCRIPTION
**Pull request by hyunilmmt**
Fixes #107.

Changes proposed.

Confirmed working under the same circumstances described in issue #107.

### CODE REVIEW

**Comment by hyunilmmt** [comment]
@fizyk Let me know what exactly you need and I will try to get those prepared. Thanks.

**Comment by fizyk** [comment]
https://github.com/ClearcodeHQ/pytest-mysql/blob/master/tests/test_executor.py

I need extended test_version_check and test_implementation

**Comment by hyunilmmt** [comment]
@fizyk The tests are extended with two extra sets of parameters, one for 5.7.23 installed using `brew`, and one for 8.0.12 installed with DMG file downloaded from MySQL's website (both on OS X). Let me know if this is adequate.

File: tests/test_executor.py [hunk 1]
```diff
@@ -22,7 +22,23 @@
             b'for debian-linux-gnu on x86_64 (Ubuntu 17.10)'
         ),
         '10.1.30'
-    )
+    ),
+    (b'mysql 8.0.12', '8.0.12'),
+    (
+        (
+            b'Ver 8.0.12'
```
Comment (fizyk) [inline tests/test_executor.py]: There seems to be exec name missing?

Comment (hyunilmmt) [reply tests/test_executor.py]: My bad. I've just added the exec name.

File: src/pytest_mysql/executor.py [hunk 2]
```diff
@@ -77,7 +77,7 @@ def version(self):
         version_output = subprocess.check_output(
             [self.mysqld, '--version']
         ).decode('utf-8')
-        return self.VERSION_RE.match(version_output).groupdict()['version']
+        return self.VERSION_RE.search(version_output).groupdict()['version']
```
Comment (fizyk) [inline src/pytest_mysql/executor.py]: funny issue is that without these changes, the tests passes as well on my machine. Is it possible that problem is somewhere else?

Comment (fizyk) [reply src/pytest_mysql/executor.py]: the only case where it helps, is when the output is two lines actually (first being empty)

Comment (hyunilmmt) [reply src/pytest_mysql/executor.py]: @fizyk Weird. It wouldn't work without these changes on my Mac at work, and neither on my co-workers.

Comment (fizyk) [reply src/pytest_mysql/executor.py]: If you could try to check what's the exact `version_output` value that would be great.

Comment (hyunilmmt) [reply src/pytest_mysql/executor.py]: @fizyk the exact values were described in the issue #107 :) I am surprised how this works differently on different systems.

**Pull request merged**